lib/models/room.js

```diff
@@ -1120,70 +1120,81 @@ function calculateRoomName(room, userId, ignoreRoomNameEvent) {
     }
 
     // get members that are NOT ourselves and are actually in the room.
-    var members = utils.filter(room.currentState.getMembers(), function(m) {
+    var otherMembers = utils.filter(room.currentState.getMembers(), function(m) {
         return (m.userId !== userId && m.membership !== "leave");
     });
+    var allMembers = utils.filter(room.currentState.getMembers(), function(m) {
+        return (m.membership !== "leave");
+    });
+    var myMemberEventArray = utils.filter(room.currentState.getMembers(), function(m) {
+        return (m.userId == userId);
+    });
+    var myMemberEvent = (
+        (myMemberEventArray.length && myMemberEventArray[0].events) ?
+            myMemberEventArray[0].events.member.event : undefined
+    );
+
     // TODO: Localisation
-    if (members.length === 0) {
-        var memberList = utils.filter(room.currentState.getMembers(), function(m) {
-            return (m.membership !== "leave");
-        });
-        if (memberList.length === 1) {
+    if (myMemberEvent && myMemberEvent.content.membership == "invite") {
+        if (otherMembers.length && otherMembers[0].events.member) {
```

Comment:
ok so I understand this better now. Could you add a comment to clarify the assumption being made?